apps/drec-api/src/pods/file/file.service.ts

```diff
@@ -41,33 +45,44 @@ export class FileService {
           filename: this.generateUniqueFilename(file.originalname),
           data: file.buffer,
           contentType: file.mimetype,
-          userId: user.id.toString(),
-          organizationId: user.organizationId?.toString(),
+          userId: user?.id?.toString(),
+          organizationId: user?.organizationId?.toString(),
           isPublic,
         });
         await entityManager.insert<File>(File, fileToStore);
 
         storedFile.push(fileToStore.id);
       }
     });
-    this.logger.debug(
-      `User ${JSON.stringify(user)} has stored ${JSON.stringify(storedFile)}`,
-    );
+    if (user) {
+      this.logger.debug(
+        `User ${JSON.stringify(user)} has stored ${JSON.stringify(storedFile)}`,
+      );
+    }
 
     return storedFile;
   }
 
-  public async get(id: string, user: ILoggedInUser): Promise<File | undefined> {
-    this.logger.debug(`User ${JSON.stringify(user)} requested file ${id}`);
+  public async get(
+    id: string,
+    user?: ILoggedInUser,
```

Comment:
it probably would be better to just create two different methods

Reply:
I simplified it a bit so it should be ok